### ISSUE TITLE
Go to definition crashes IDE when used in comptime emit view 

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1542]
Tested with: Nightly 04/22/2022
Code to reproduce
```cs
struct FooAttribute : Attribute, IComptimeTypeApply
{
    [Comptime]
    public void ApplyToType(Type type)
    {
        String fields = new .();
        for (let f in type.GetFields(.Public | .NonPublic | .Instance))
        {
            fields.AppendF($"\tConsole.WriteLine(\"{f.Name} {{0}}\", {f.Name});\n\t");
        }
        Compiler.EmitTypeBody(type,
            scope $$"""
                public void Print()
                {
                {{fields}}
                }
            """);
    }
}

[Foo]
struct Bar 
{
    public int x;
    internal int qwe;
    public float Val { get; set mut; }
    public this()
    {
        x = 1;
        qwe = 10;
        Val = 50;
    }
}


public static void Main()
{
    Bar x = .();
    x.Print();
}
```
The code is not actually necessary, it will crash with any emitted code. But I included it because i have a question, the above code doesn't compile because of the `public float Val { get; set mut; }` it errors out in generated code with "prop__Val is inaccessible due to its protection level" is there a way to get around that (i tried `[Friend]` which didn't help) or should i just ignore fields with `prop__` prefix?

**Comment by Fusioon** [comment]
I just realized that i can simply remove the `prop__` prefix when needed and it will work.

**Comment by bfiete** [comment]
Fixed at e30744836375d93eb32faa0ef3ba261e1db2800d

Also you can directly access `prop__` autoprop backing fields now, too.
